enforcer/resources/owner_store.py

```diff
@@ -0,0 +1,126 @@
+import logging
+import threading
+import time
+from typing import Dict, Any, Optional, List
+
+from enforcer.env_vars import REPLICA_SET_CLEANUP_INTERVAL, REPLICA_SET_DELETION_WAIT
+from enforcer.model import PodOwner, RsOwner
+from enforcer.resources.kubernetes_resource_loader import KubernetesResourceLoader
+
+
+class OwnerStore:
+
+    def __init__(self):
+        self.rs_owners = self._load_rs_owners()
+        self._rs_owners_lock = threading.Lock()
+        self.cleanup_interval = REPLICA_SET_CLEANUP_INTERVAL
+        self._stop_event = threading.Event()
+        self._cleanup_thread = threading.Thread(target=self._periodic_cleanup, daemon=True)
+        self._cleanup_thread.start()
+
+    def _rs_key(self, rs_name: str, namespace: str) -> str:
+        return f"{namespace}/{rs_name}"
+
+    def _load_rs_owners(self) -> Dict[str, RsOwner]:
+        replica_sets_owners: List[RsOwner] = KubernetesResourceLoader.load_replicasets()
+        rs_owners: Dict[str, RsOwner] = {}
+        for owner in replica_sets_owners:
+            rs_owners[self._rs_key(owner.rs_name, owner.namespace)] = owner
+        return rs_owners
+
+    @staticmethod
+    def get_pod_name(metadata: Dict[str, Any]) -> str:
+        # if the pod's name is randomized, the name is under generateName
+        return metadata.get("name") or metadata.get("generateName")
```

Comment:
I'm not sure I understand this part. I checked a pod created by a replicaset and it had both name and generateName. Futhermore, you only use this once below for self-owned pods. I don't understand why we need generateName here and when it is different.

Reply:
I use it in 2 places
on an admission review, for a pod that belongs to a replica set, it has only generateName (no name)